src/caustics/tests.py

```diff
@@ -32,18 +37,22 @@ def _test_simulator_runs():
         name="lenslight", x0=0.0, y0=0.01, q=0.7, phi=pi / 4, n=3.0, Re=0.7, Ie=1.0
     )
 
-    psf = gaussian(0.05, 11, 11, 0.2, upsample=2)
+    psf = gaussian(0.05, 11, 11, 0.2, upsample=2, device=TORCH_DEVICE)
 
+    z_s = torch.tensor(2.0, device=TORCH_DEVICE)
     sim = Lens_Source(
         lens=lensmass,
         source=source,
         pixelscale=0.05,
         pixels_x=50,
         lens_light=lenslight,
         psf=psf,
-        z_s=2.0,
+        z_s=z_s,
     )
 
+    # Send to device
+    sim = sim.to(TORCH_DEVICE)
+
     assert torch.all(torch.isfinite(sim()))
```

Comment:
Looks like the failure that I pointed out to traceback is on this line, where `sim()` is called.. I had thought that I mitigate this by calling `sim.to` on L54. I must be missing something else.

Reply:
You are right, the `.to` method should have handled this. I'll take a look

Reply:
I think I found the issue, I pushed it to dev. Can you try updating to the latest dev branch?